Sledge.Packages/Wad/WadPackage.cs

```diff
@@ -136,8 +141,16 @@ private void SetEntryData(WadEntry e, BinaryReader br)
                     var num = (int)(width * height);
                     var skipMapData = (num / 4) + (num / 16) + (num / 64);
                     br.BaseStream.Position += 16 + num + skipMapData; // Skip mipmap offsets, texture data, mipmap texture data
-                    paletteSize = br.ReadUInt16();
-                    paletteDataOffset = br.BaseStream.Position;
+                    if (e.Type == WadEntryType.QuakeTexture)
```

Comment:
I think it'd probably be cleaner to branch in the switch instead of here, maybe something like this?

```
case WadEntryType.Texture:
    textureDataOffset = MoveToTextureData(br, out width, out height);
    paletteSize = br.ReadUInt16();
    paletteDataOffset = br.BaseStream.Position;
    break;
case WadEntryType.QuakeTexture:
    textureDataOffset = MoveToTextureData(br, out width, out height);
    paletteSize = 256;
    paletteDataOffset = 0;
    break;
    
...

private long MoveToTextureData(BinaryReader br, out uint width, out uint height)
{
    br.BaseStream.Position += 16; // Skip name
    width = br.ReadUInt32();
    height = br.ReadUInt32();
    var textureDataOffset = br.BaseStream.Position + 16;
    var num = (int)(width * height);
    var skipMapData = (num / 4) + (num / 16) + (num / 64);
    br.BaseStream.Position += 16 + num + skipMapData; // Skip mipmap offsets, texture data, mipmap texture data
    return textureDataOffset;
}
```